src/kudu/tserver/simple_tablet_manager.cc

```diff
@@ -117,36 +122,208 @@ TSTabletManager::TSTabletManager(TabletServer* server)
+    }
+    set<string> peer_addrs_from_disk;
+    for (const auto& p : cstate.committed_config().peers()) {
+      HostPort hp;
+      RETURN_NOT_OK(HostPortFromPB(p.last_known_addr(), &hp));
+      peer_addrs_from_disk.insert(hp.ToString());
+    }
+    vector<string> symm_diff;
+    std::set_symmetric_difference(peer_addrs_from_opts.begin(),
+                                  peer_addrs_from_opts.end(),
+                                  peer_addrs_from_disk.begin(),
+                                  peer_addrs_from_disk.end(),
+                                  std::back_inserter(symm_diff));
+    if (!symm_diff.empty()) {
+      string msg = Substitute(
+          "on-disk master list ($0) and provided master list ($1) differ. "
+          "Their symmetric difference is: $2",
+          JoinStrings(peer_addrs_from_disk, ", "),
+          JoinStrings(peer_addrs_from_opts, ", "),
+          JoinStrings(symm_diff, ", "));
+      return Status::InvalidArgument(msg);
+    }
+  }
+
+  RETURN_NOT_OK(SetupRaft());
+  return Status::OK();
+}
+
+Status TSTabletManager::CreateNew(FsManager *fs_manager) {
+  RaftConfigPB config;
+  if (server_->opts().IsDistributed()) {
+    RETURN_NOT_OK_PREPEND(CreateDistributedConfig(server_->opts(), &config),
+                          "Failed to create new distributed Raft config");
+  } else {
+    config.set_obsolete_local(true);
+    config.set_opid_index(consensus::kInvalidOpIdIndex);
+    RaftPeerPB* peer = config.add_peers();
+    peer->set_permanent_uuid(fs_manager->uuid());
+    peer->set_member_type(RaftPeerPB::VOTER);
+  }
+
+  RETURN_NOT_OK_PREPEND(cmeta_manager_->Create(kSysCatalogTabletId, config, consensus::kMinimumTerm),
+                        "Unable to persist consensus metadata for tablet " + kSysCatalogTabletId);
+
+  return SetupRaft();
+}
+
+Status TSTabletManager::CreateDistributedConfig(const TabletServerOptions& options,
+                                                RaftConfigPB* committed_config) {
+  DCHECK(options.IsDistributed());
+
+  RaftConfigPB new_config;
+  new_config.set_obsolete_local(false);
+  new_config.set_opid_index(consensus::kInvalidOpIdIndex);
+
+  // Build the set of followers from our server options.
+  for (const HostPort& host_port : options.tserver_addresses) {
+    RaftPeerPB peer;
+    HostPortPB peer_host_port_pb;
+    RETURN_NOT_OK(HostPortToPB(host_port, &peer_host_port_pb));
+    peer.mutable_last_known_addr()->CopyFrom(peer_host_port_pb);
+    peer.set_member_type(RaftPeerPB::VOTER);
+    new_config.add_peers()->CopyFrom(peer);
+  }
+
+  // Now resolve UUIDs.
+  // By the time a SysCatalogTable is created and initted, the masters should be
+  // starting up, so this should be fine to do.
+  DCHECK(server_->messenger());
+  RaftConfigPB resolved_config = new_config;
+  resolved_config.clear_peers();
+  for (const RaftPeerPB& peer : new_config.peers()) {
+    if (peer.has_permanent_uuid()) {
```

Comment:
I don't think the if block would ever execute because we haven't set the uuid in the protobufs in 201-208.

Reply:
yes, you are right. I think this was kept just for safety in the original code.